sycl-fusion/common/include/DynArray.h

```diff
@@ -84,6 +85,28 @@ template <typename T> class DynArray {
   }
 };
 
+///
+/// String-like class that owns its character storage and provides implicit
+/// conversions from and to C strings.
+class DynString {
+public:
+  DynString(const char *Str) {
+    size_t Len = std::strlen(Str);
+    Chars = DynArray<char>(Len + 1);
+    std::copy(Str, Str + Len, Chars.begin());
+    Chars[Len] = '\0';
+  }
```

Comment:
```suggestion
  explicit DynString(const char *Str) : Chars{std::strlen(Str) + 1} {
    std::copy(Str, Str + Chars.size(), Chars.begin());
    Chars[Len] = '\0';
  }
```
Avoid default initialization